src/main/java/org/graylog2/gelfclient/transport/GelfHttpTransport.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2018 Graylog, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.graylog2.gelfclient.transport;
+
+import io.netty.bootstrap.Bootstrap;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioSocketChannel;
+import io.netty.handler.codec.http.HttpClientCodec;
+import io.netty.handler.codec.http.HttpContentDecompressor;
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslContextBuilder;
+import io.netty.handler.ssl.util.InsecureTrustManagerFactory;
+import org.graylog2.gelfclient.GelfConfiguration;
+import org.graylog2.gelfclient.encoder.GelfHttpEncoder;
+import org.graylog2.gelfclient.encoder.GelfMessageJsonEncoder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A {@link GelfTransport} implementation that uses HTTP(S) to send GELF messages.
+ * <p>This class is thread-safe.</p>
+ */
+public class GelfHttpTransport extends AbstractGelfTransport {
+    private static final Logger LOG = LoggerFactory.getLogger(GelfHttpTransport.class);
+
+    /**
+     * Creates a new TCP GELF transport.
+     *
+     * @param config the GELF client configuration
+     */
+    public GelfHttpTransport(GelfConfiguration config) {
+        super(config);
+    }
+
+    @Override
+    protected void createBootstrap(final EventLoopGroup workerGroup) {
+        final Bootstrap bootstrap = new Bootstrap();
+        final GelfSenderThread senderThread = new GelfSenderThread(queue, config.getMaxInflightSends());
+        senderThreadReference.set(senderThread);
+
+        bootstrap.group(workerGroup)
+                .channel(NioSocketChannel.class)
+                .option(ChannelOption.CONNECT_TIMEOUT_MILLIS, config.getConnectTimeout())
+                .option(ChannelOption.TCP_NODELAY, config.isTcpNoDelay())
+                .option(ChannelOption.SO_KEEPALIVE, config.isTcpKeepAlive())
+                .remoteAddress(config.getRemoteAddress())
```

Comment:
This is not taking the `GelfConfiguration#getUri()` configuration option into account. (see comment in `GelfConfiguration.java`)